integration-test/src/test/java/com/datastrato/gravitino/integration/test/container/BaseContainer.java

```diff
@@ -124,7 +125,33 @@ public void start() {
     container.start();
   }
 
-  protected abstract boolean checkContainerStatus(int retryLimit);
+  protected abstract boolean checkContainerStatus(int timeoutMillis);
+
+  protected boolean checkContainerStatusWithRetry(int timeoutMillis, BooleanSupplier... checker) {
+    int nRetry = 0;
+    int sleepTimeMillis = 5_000;
+    int totalSleepTimeMillis = 0;
+
+    // retry until timeout
+    for (BooleanSupplier check : checker) {
+      while (totalSleepTimeMillis < timeoutMillis) {
+        if (check.getAsBoolean()) {
+          break;
+        }
+
+        sleepTimeMillis = sleepTimeMillis * (int) Math.pow(2, nRetry++);
+        sleepTimeMillis = Math.min(sleepTimeMillis, timeoutMillis - totalSleepTimeMillis);
+        LOG.info("retrying after {}ms (retry count: {})", sleepTimeMillis, nRetry);
+        try {
+          Thread.sleep(sleepTimeMillis);
+        } catch (InterruptedException e) {
+          // ignore
+        }
+        totalSleepTimeMillis += sleepTimeMillis;
+      }
+    }
+    return totalSleepTimeMillis < timeoutMillis;
```

Comment:
If the check procedure is successful, I believe we should use the value of "check.getAsBoolean()."

Reply:
Here outside the block of FOR loop, if we use `check.getAsBoolean()`, an additional check is performed.

BTW, there are multiple checkers here. Which checker's method `check.getAsBoolean()` are you referring to?